### ISSUE TITLE
deadlock in Aircrack-ng when reading a corrupt capture file

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#25]
**Reported by misterx on 22 Dec 2006 22:19 UTC**

In read_thread() in aircrack-ng.c, there is a possibility of a deadlock at the mutex lock (mx_apl) due to a continue which doesn't unlock the mutex.

This manifests itself as a hang when reading from a capture file (which contains a packet formed in such a way that it triggers the problem). The output looks like this:

```
aircrack-ng file.cap
Opening file.cap
Reading packets, please wait...
```

A possible fix:
```
--- aircrack-ng-0.6/src/aircrack-ng.c   2006-06-22 20:53:57.000000000 -0700
+++ aircrack-ng-0.6_modified/src/aircrack-ng.c  2006-07-04 20:54:47.000000000 -0700
@@ -845,7 +845,7 @@
                ap_cur->crypt = 3;               /* set WPA */

                if( st_cur == NULL )
-                       continue;
+                       goto unlock_mx_apl;

                /* frame 1: Pairwise == 1, Install == 0, Ack == 1, MIC == 0 */
```

See http://tinyshell.be/aircrackng/forum/index.php?topic=322.0

**Comment by aircrack-ng** [comment]
**Comment by misterx on 22 Dec 2006 22:19 UTC**

More checks must be done when reading a pcap file to avoid such problems

**Comment by aircrack-ng** [comment]
**Comment by misterx on 22 Dec 2006 22:19 UTC**

(In #41) Should fix a deadlock when reading a corrupt file (Closes: #25)
